doc/api-backend.rst

```diff
@@ -114,6 +115,164 @@ IXMP4Backend
 .. automodule:: ixmp.util.ixmp4
    :members:
 
+Differences between JDBCBackend and IXMP4Backend
+------------------------------------------------
+
+.. NOTE What do we do about major difference such as Run being the new Scenario, IndexSet and Table vs Set, etc?
+.. Functionally, all things are equivalent, but these *are* changes we'd like people to adjust to eventually.
+
+While we strive to make IXMP4Backend as functionally compatible to JDBCBackend as possible,
+there are several differences remaining.
+Some of them are intentional,
+so are not expected to change,
+while others may still be addressed.
+
+CLI integration
+^^^^^^^^^^^^^^^
+
+Some CLI tests do not work on ixmp4 yet, but all are supposed to.
+Be careful when using ``ixmp platform list`` 
+and ``ixmp solve``
+on the command line.
+
+Regions
+^^^^^^^
+
+On JDBCBackend, each region can have a "parent" and a "hierarchy"
+and may have a "synonym" defined for itself.
+Notably, when a region is part of a "hierarchy", a "parent" must be defined.
+
+On ixmp4, each region must be part of a "hierarchy", 
+but does not have either "parent" nor "synonym".
+
+For backward compatibility, IXMP4Backend accepts the same parameters for region-related function as JDBCBackend.
```

Comment:
In general, it will be better to write this from the perspective of a user of ixmp or someone who is using ixmp through message_ix. They may not know/care about the background classes, but will need to be aware that `Platform.add_region()` is affected.

This will also help them search and find this docs section if they try to use that particular method and something unexpected happens.

Reply:
I'll join in on the PR to copyedit this docs section a bit.